core/tests/protractor/cacheSlugs.js

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end tests for cache slugs.
+ */
+
+var general = require('../protractor_utils/general.js');
+
+var ERROR_PAGE_URL_SUFFIX = '/console_errors';
+
+var checkConsoleErrorsExist = function(errorsToFind) {
+  browser.manage().logs().get('browser').then(function(browserLogs) {
+    var errorsExpected = [];
+    for (var i = 0; i < browserLogs.length; i++) {
+      var errorPresent = false;
+      for (var j = 0; j < errorsToFind.length; j++) {
+        if (browserLogs[i].message.match(errorsToFind[j])) {
+          errorPresent = true;
+        }
+      }
+      expect(errorPresent).toBe(true);
```

Comment:
This is still not testing what you want -- please think through the various cases. What if browserLogs is empty? This test should error, but it doesn't.

Basically I would suggest passing in what you _expect_ the browser logs to be and just comparing them one-by-one to the actual browser logs, erroring if there's any mismatch at all. That's simplest, and easy to understand.